### PR TITLE
Implement bandit.evaluate

### PR DESCRIPTION
**Pull request by MaxHalford**
I called it `evaluate` to be consistent with the existing method in the `time_series` module.

I'll make use of this function in a future user guide to reproduce the examples from Sutton and Barto :)

@AdilZouitine can you take a quick pass? Not sure I'm using/copying Gym environments correctly.

### CODE REVIEW

File: river/bandit/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    pull_func
+        A function that takes a policy and an environment as arguments and returns the arm that was
+        pulled. This function is called at each time step for each policy. This is required because
+        there is no standard way to pull an arm in Gym environments.
+    reward_stat
+        A univariate statistic to keep track of the rewards. This statistic will be reset before
+        each episode. Note that this is not the same as the reward object used by the policies.
+        It's just a statistic to keep track of each policy's performance. If `None`, `stats.Sum`
+        is used.
+    n_episodes
```
Comment (AdilZouitine) [inline river/bandit/evaluate.py:42]: I think it is the best choice, and `Sum` will be used 99% of the time for sure

File: river/bandit/evaluate.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+        while not done:
+            terminated, truncated = False, False
+            for i, (_policy, _env, _reward_stat) in enumerate(
+                zip(episode_policies, episode_envs, episode_reward_stats)
+            ):
+
+                action = pull_func(_policy, _env)
+                observation, reward, terminated, truncated, info = _env.step(action)
+                _reward_stat.update(reward)
```
Comment (AdilZouitine) [inline river/bandit/evaluate.py:141]: You have several times the same environment instance sequentially.

In this case, one of the environments may end before the others.
This is not handled; the finished environment is often reset, and the steps are continued.

**Review by AdilZouitine — COMMENTED** [review]
The Pr is excellent! There is a point that concerns me is the fact of not managing the rest of the episodes independently because it is possible that the episodes do not end at the same time

File: river/bandit/evaluate.py (hunk 2 above)
Comment (MaxHalford) [reply river/bandit/evaluate.py:141]: Ah yes good point. I've only worked with envs that end after a fixed number of episodes. But you're right, we should handle the general case.

**Pull request merged**